src/pages/index.js

```diff
@@ -39,38 +39,148 @@ const profFormPopup = new PopupWithForm(
   ".modal_type_profile",
   handleProfileSubmit
 );
+const delCardFormPopup = new PopupWithForm(
+  ".modal_type_delete-card",
+  handleDelCardSubmit
+);
+const editAviFormPopup = new PopupWithForm(
+  ".modal_type_edit-avi",
+  handleEditAviSubmit
+);
 const profUserInfo = new UserInfo({
   userNameSelec: ".profile__name",
   descrSelec: ".profile__title",
+  aviSelec: ".profile__avatar",
+});
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "1e41f1e9-9859-4a6e-b008-27b738619ee0",
+    "Content-Type": "application/json",
+  },
 });
+const cardSection = new Section({ renderer: createCard }, ".gallery");
+//INITIALLY SET USER INFO
+api
+  .getInitialUserInfo()
+  .then((data) => {
+    profUserInfo.setUserInfo(data);
+  })
+  .catch((err) => console.error(err));
 
 // POPULATE INITIAL CARDS
 export function createCard(data) {
-  const cardElement = new Card(data, "#card-template", handleCardEnlargement);
+  const cardElement = new Card(
+    data,
+    "#card-template",
+    handleCardEnlargement,
+    handleTrashClick,
+    sendLikeReq
+  );
   return cardElement.render();
 }
 
-section.renderItems();
+api
+  .getInitialCards()
+  .then((data) => {
+    data.reverse();
+
+    cardSection.renderItems(data);
+  })
+  .catch((err) => console.error(err));
 
 // ENLARGE CARD IMAGE
 function handleCardEnlargement(evt) {
   enlrgImgPopup.open(evt);
 }
 
 // SUBMIT NEW CARD MODAL
-export function handleNewCardFormSubmit(newCardData) {
-  section.addItem(createCard(newCardData));
-
-  newCardFormPopup.close();
-  newCardModalForm.reset();
+function handleNewCardFormSubmit({ name, link }) {
+  newCardFormPopup.popup.querySelector(".modal__submit-button").textContent =
+    "Creating...";
 
-  newCardFormValidator.resetValidation();
+  api
+    .postCards(name, link)
+    .then((data) => {
+      cardSection.renderItems([data]);
+      newCardModalForm.reset();
+      newCardFormValidator.resetValidation();
```

Comment:
> I need to create the `disableButton` method

`resetValidation` **hides** errors which can't be when you submit the form, so, you need to make another method `disableButton` for disabling the button (without hiding errors)


It's a waste of resources in removing errors which don't exist 100%

So, please, make method `disableButton` like that


```js

  disableButton() {
    this._toggleSubmitButton("disable");
  }
```


And then call it here